src/main/java/com/sourcery/gymapp/backend/userProfile/service/UserProfileService.java

```diff
@@ -0,0 +1,61 @@
+package com.sourcery.gymapp.backend.userProfile.service;
+
+import com.sourcery.gymapp.backend.userProfile.dto.UserProfileDto;
+import com.sourcery.gymapp.backend.userProfile.exception.UserNotFoundException;
+import com.sourcery.gymapp.backend.userProfile.exception.UserProfileNotFoundException;
+import com.sourcery.gymapp.backend.userProfile.model.UserProfile;
+import com.sourcery.gymapp.backend.userProfile.repository.UserProfileRepository;
+import com.sourcery.gymapp.backend.userProfile.mapper.UserProfileMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class UserProfileService {
+
+    private final AuditorAware<UUID> auditorAware;
+    private final UserProfileRepository userProfileRepository;
+    private final UserProfileMapper userProfileMapper;
+
+    public UserProfileDto getUserProfile() throws UserNotFoundException, UserProfileNotFoundException {
+
+        UserProfile userProfile = userProfileRepository
+                .findUserProfileByUserId(
+                        auditorAware
+                                .getCurrentAuditor()
+                                .orElseThrow(UserNotFoundException::new));
+
+        if (userProfile==null)
+        {
+            throw new UserProfileNotFoundException();
+        }
+
+        return userProfileMapper.toDto(userProfile);
+    }
+
+    @Transactional
+    public UserProfileDto updateUserProfile(UserProfileDto dto) {
+        UserProfile entity = userProfileRepository.save(userProfileMapper.toEntity(dto));
+        return  userProfileMapper.toDto(entity);
+    }
+
+    @Transactional
+    public UserProfileDto deleteUserProfile() throws UserNotFoundException, UserProfileNotFoundException {
+
+        UserProfile userProfile =  userProfileRepository.findUserProfileByUserId(
+                auditorAware
+                        .getCurrentAuditor()
+                .orElseThrow(UserNotFoundException::new));
+
+        if (userProfile == null) {
+                throw new UserProfileNotFoundException();
```

Comment:
I don't know if we need this check, if we had `.orElseThrow(UserNotFoundException::new)` above
But maybe I don't catch something

Reply:
the thing is there could be no user profile even though user is registered, so yeah, we need it

Reply:
ok, got it